### ISSUE TITLE
TCXO Not Detected (Firmware Version:2023.01.01)

### ISSUE THREAD

**Issue by muuzoo** [greatscottgadgets/hackrf#1351]
### What type of issue is this?

transient - occurring only once

### What issue are you facing?

I have two Hackrf One devices. The TCXO module detected on one device but not the other. 

Information about the working device with TCXO:

```
hackrf_info 
hackrf_info version: git-13dacd81
libhackrf version: git-13dacd81 (0.8)
Found HackRF
Index: 0
Serial number: 0000000000000000579c61dc232a609f
Board ID Number: 2 (HackRF One)
Firmware Version: local-57ed3eb (API:1.04)
Part ID Number: 0xa000cb3c 0x005d4766

hackrf_debug --si5351c -n 0 -r
[  0] -> 0x01

hackrf_clock -a
MS0:
	clock control = Up, Frac Mode, PLL src A, MULTISYNTH SELF, 8 mA
	p1 = 4608
	p2 = 0
	p3 = 0
	output divider = 2
MS1:
	clock control = Up, Int Mode, PLL src A, Inverted, MULTISYNTH 0 4, 2 mA
	p1 = 4608
	p2 = 0
	p3 = 1
	Output (800Mhz PLL): 20.0000000000 Mhz
	output divider = 1
MS2:
	clock control = Up, Int Mode, PLL src A, MULTISYNTH 0 4, 2 mA
	p1 = 4608
	p2 = 0
	p3 = 1
	Output (800Mhz PLL): 20.0000000000 Mhz
	output divider = 1
MS3:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1 = 0
	p2 = 0
	p3 = 0
	output divider = 1
MS4:
	clock control = Up, Int Mode, PLL src A, Inverted, MULTISYNTH SELF, 6 mA
	p1 = 2048
	p2 = 0
	p3 = 1
	Output (800Mhz PLL): 40.0000000000 Mhz
	output divider = 1
MS5:
	clock control = Up, Int Mode, PLL src A, MULTISYNTH SELF, 4 mA
	p1 = 2048
	p2 = 0
	p3 = 1
	Output (800Mhz PLL): 40.0000000000 Mhz
MS6:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1_int = 0
	output divider = 1
MS7:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1_int = 0
	output divider = 1
```

Information about the  failed device:

```
hackrf_info 
hackrf_info version: git-13dacd81
libhackrf version: git-13dacd81 (0.8)
Found HackRF
Index: 0
Serial number: 0000000000000000f75461dc278e3fc3
Board ID Number: 4 (HackRF One)
Firmware Version: 2023.01.1 (API:1.07)
Part ID Number: 0xa000cb3c 0x005d435d
Hardware Revision: r9
Hardware does not appear to have been manufactured by Great Scott Gadgets.
Hardware supported by installed firmware:
    HackRF One

hackrf_debug --si5351c -n 0 -r
[  0] -> 0x51

hackrf_clock -a
MS0:
	clock control = Up, Int Mode, PLL src A, MULTISYNTH SELF, 6 mA
	p1 = 2048
	p2 = 0
	p3 = 1
	Output (800Mhz PLL): 40.0000000000 Mhz
	output divider = 1
MS1:
	clock control = Up, Frac Mode, PLL src A, MULTISYNTH SELF, 4 mA
	p1 = 4608
	p2 = 0
	p3 = 0
	output divider = 1
MS2:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1 = 0
	p2 = 0
	p3 = 0
	output divider = 1
MS3:
	clock control = Down, Frac Mode, PLL src A, XTAL, 2 mA
	p1 = 0
	p2 = 0
	p3 = 0
	output divider = 1
MS4:
	clock control = Down, Frac Mode, PLL src A, XTAL, 2 mA
	p1 = 0
	p2 = 0
	p3 = 0
	output divider = 1
MS5:
	clock control = Down, Frac Mode, PLL src A, XTAL, 2 mA
	p1 = 0
	p2 = 0
	p3 = 0
	output divider = 1
MS6:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1_int = 0
	output divider = 1
MS7:
	clock control = Down, Int Mode, PLL src A, XTAL, 2 mA
	p1_int = 0
	output divider = 1
```

### What are the steps to reproduce this?

Just install external TCXO module  (ex: Tiny TCXO 0.5 PPM) on connector P22

**Comment by Caian** [comment]
I'm also experiencing this issue with a newly bought TXCO and HackRF One (Same board ID number 4 and FW)

**Comment by dmaltsiniotis** [comment]
What TCXO is in use here? It's been found recently that some popular TCXO's on the market do not conform to the the proper voltage specifications the HackRF One requires. Previous board revisions have been more forgiving about misbehaving TCXO's but more recent revisions require TCXO's to perform within spec.

**Comment by Caian** [comment]
> What TCXO is in use here? It's been found recently that some popular TCXO's on the market do not conform to the the proper voltage specifications the HackRF One requires. Previous board revisions have been more forgiving about misbehaving TCXO's but more recent revisions require TCXO's to perform within spec.

It's a generic TCXO that the seller sent with HackRF. I'll see if I can get better one.

**Comment by muuzoo** [comment]
I tried two different TCXOs on r9 board and it didn't work. I also have Portapack H2 and the TCXO on it didn't work either with r9 board.

Here is the image of TCXOs:

![portapack](https://github.com/greatscottgadgets/hackrf/assets/9133424/c43dd2b0-ba41-4a3e-bf14-72f1dc87b745)

![tcxo_front](https://github.com/greatscottgadgets/hackrf/assets/9133424/e8e46d04-3c6a-420e-9e10-9df13dc7eef1)

![tcxo_back](https://github.com/greatscottgadgets/hackrf/assets/9133424/d2016063-a808-4999-9efa-98dc759fd3fb)

**Comment by straithe** [comment]
Do you have purchasing links to the TCXOs? I'll need to see specs for them. You'll want to make sure that they are 3.3 V.

**Comment by sreevaishnav-git** [comment]
@straithe both the TCXO's OP posted are rated for 3.3V. I am also experiencing the same problem.

**Comment by sreevaishnav-git** [comment]
@straithe I'm also experiencing the same issue. I have a tiny TCXO (left blue one on the above pic). [Link](https://robosavvy.co.uk/hackrf-one-tiny-tcxo-10mhz-0-5ppm-tcxo-module.html)

**Comment by jolyfra** [comment]
I see the same behavior with my v9 in 2023.01.1 but there is a mismatch between hackrf_clock -i => Ok and hackrf_debug --si5351c -n 0 -r => Ko

$ hackrf_clock -i
CLKIN status: clock signal detected

$ hackrf_debug --si5351c -n 0 -r
[  0] -> 0x51

Is there any other way to verify TCXO detection?

It is stated in the HackRF release note that on v9 the Si5351C has been replaced by a Si5351A.
Does the current firmware really take this into account?

https://hackrf.readthedocs.io/en/latest/list_of_hardware_revisions.html
https://www.skyworksinc.com/en/Products/Timing/CMOS-Clock-Generators

**Comment by straithe** [comment]
> @straithe I'm also experiencing the same issue. I have a tiny TCXO (left blue one on the above pic). [Link](https://robosavvy.co.uk/hackrf-one-tiny-tcxo-10mhz-0-5ppm-tcxo-module.html)


I'm not seeing whether this TCXO is 3.3 V. Can you confirm with the reseller?

**Comment by straithe** [comment]
> I see the same behavior with my v9 in 2023.01.1 but there is a mismatch between hackrf_clock -i => Ok and hackrf_debug --si5351c -n 0 -r => Ko
> 
> $ hackrf_clock -i CLKIN status: clock signal detected
> 
> $ hackrf_debug --si5351c -n 0 -r [ 0] -> 0x51
> 
> Is there any other way to verify TCXO detection?
> 
> It is stated in the HackRF release note that on v9 the Si5351C has been replaced by a Si5351A. Does the current firmware really take this into account?
> 
> https://hackrf.readthedocs.io/en/latest/list_of_hardware_revisions.html https://www.skyworksinc.com/en/Products/Timing/CMOS-Clock-Generators

There are some TCXO modules out there which are not compatible with the R9 hardware since they do not output the necessary voltage. We've always specified that a 3.3.V signal is required. Some modules were apparently being sold that didn't provide enough voltage, yet worked by chance with earlier HackRF One revisions.

**Comment by straithe** [comment]
@muuzoo have you been able to find a TCXO with 3.3 V?

**Comment by Caian** [comment]
> Do you have purchasing links to the TCXOs? I'll need to see specs for them. You'll want to make sure that they are 3.3 V.

I bough it in a bundle with HackRF https://www.aliexpress.com/item/32972753170.html

**Comment by straithe** [comment]
> > Do you have purchasing links to the TCXOs? I'll need to see specs for them. You'll want to make sure that they are 3.3 V.
> 
> I bough it in a bundle with HackRF https://www.aliexpress.com/item/32972753170.html

Please contact that seller to verify that the TCXO is 3.3 V.

**Comment by sizzlik** [comment]
on my devices i noticed that none of the small TCXO like the nooelec work on R9 clones, no matter wich seller, but the bulky silver ones do.

**Comment by straithe** [comment]
I am closing this issue as there has not been a response from the original poster in over a month. If you are not the original poster and still need assistance, please open a new issue.
